engine/class_modules/paladin/sc_paladin.cpp

```diff
@@ -2024,6 +2025,18 @@ double paladin_t::composite_melee_crit_chance() const
   return h;
 }
 
+double paladin_t::composite_player_target_multiplier ( player_t* target, school_e school ) const
+{
+  paladin_td_t* td = get_target_data( target );
+  double cptm = player_t::composite_player_target_multiplier( target, school );
+  if ( school == SCHOOL_HOLY && td -> debuff.vengeful_shock -> up() )
```

Comment:
Surprising that this is not a whitelist, although there probably aren't many generic Holy damage effects around (Light's Judgment maybe?).

Reply:
LJ was the main one I was thinking of. Not sure if there's any trinks or suchlike.

Reply:
This check should be `dbc::is_school( school, SCHOOL_HOLY )` as far as I can tell. Otherwise it will miss composite schools like `HOLYFIRE`, `HOLYSTRIKE` and the like. Not sure if paladins themselves have spells with such but there were trinkets with such in the past.